### PR TITLE
Various fixes to make the chromecast behave better

### PR DESCRIPTION
**Pull request by hawken93**
I still have problems that it doesn't select or change subs properly. AC3 still not working, see https://issuetracker.google.com/issues/174837344

### CODE REVIEW

**Comment by hawken93** [comment]
This one is on top of my other PR.
The main changes are:
- fix the subtitle track if-block logic error
- pick different API urls than the /emby/ ones and lowercased the directstream /Videos/ endpoint
- cleaned up the call into deviceProfile, made arguments mandatory. Moved ac-3 check into codecSupportHelper.
- fixed the bug that causes it to always assume 6 channels available (`any positive number` ? 6 : 2) == 6
- prepped for AC-3 and E-AC-3 support, but made workarounds to disable them for now.
- Made a PlayNow command handler, shouldn't rely on the default handler. Increased visibility of default handler usage.
- Prolong or disable dev mode timeout (I think I'll completely disable it)
- removed some unused constructor arguments
- moved some context stuff into maincontroller to try making sure that the receiver context isn't instantiated everywhere, goal is to pass one instance around.
- Simplified getMaxBitrate and made a sane fallback for it; set StaticBitrate as well because the server seems to assume that the available bitrate is always too low

I still have issues with:
- Instability, it likes to quit on you
- Subtitle change during playback crashes it
- That API queuing is not being used
- AC-3 still not supported; pending my opened issue over at google

File: src/helpers.js [hunk 1]
```diff
@@ -421,7 +421,7 @@ export function createStreamInfo(item, mediaSource, startPosition) {
         let subStreamCodec = subtitleStream.Codec.toLowerCase();
         if (subStreamCodec !== 'vtt' &&
             subStreamCodec !== 'webvtt' &&
-            (subStreamCodec === 'srt' && subtitleStream.DeliveryMethod !== "External")) {
+            !(subStreamCodec === 'srt' && subtitleStream.DeliveryMethod !== "External")) {
```
Comment (YouKnowBlom) [inline src/helpers.js:424]: Have you tested this with external srts? I think this change is breaking that

Comment (YouKnowBlom) [reply src/helpers.js:424]: Wait no, I'm being stupid here hehe

Comment (hawken93) [reply src/helpers.js]: I can invert this logic so it'll be human readable

**Comment by hawken93** [comment]
text style change --> see commit message, I decided to write it out in there.

**Comment by hawken93** [comment]
The subtitle change doesn't always work, I was thinking to push settings when the app starts instead, and maybe again when transferring the settings if there is a change to them.

**Comment by YouKnowBlom** [comment]
Oh yeah we've been talking about doing that a while back and I definitely agree 👍

File: src/components/maincontroller.js [hunk 2]
```diff
@@ -694,14 +678,30 @@ export function setTextTrack(index) {
         if (subtitleTrack) {
             textTracksManager.setActiveByIds([subtitleTrack.trackId]);
             var subtitleAppearance = window.subtitleAppearance;
+
+            // https://developers.google.com/cast/docs/reference/chrome/chrome.cast.media.TextTrackStyle
+            var textTrackStyle = new cast.framework.messages.TextTrackStyle();
```
Comment (hawken93) [inline src/components/maincontroller.js:683]: this stuff doesn't work right now because I don't think it necessarily gets called

Comment (hawken93) [reply src/components/maincontroller.js]: removed from the PR

**Comment by hawken93** [comment]
Subtitle style fix removed from this PR, I need to rethink this part..

**Comment by hawken93** [comment]
I added a debug statement because the server address seems to get nulled sometimes.
It would be nice to see if that happens through the setter or happens due to something else.

**Review by YouKnowBlom — APPROVED** [review]

**Pull request merged**